src/main/host/descriptor/regular_file.c

```diff
@@ -531,11 +531,11 @@ ssize_t regularfile_preadv2(RegularFile* file, const Host* host, const struct io
 ssize_t regularfile_write(RegularFile* file, const void* buf, size_t bufSize) {
     MAGIC_ASSERT(file);
 
-    if (!_fd_isValid(_regularfile_getOSBackedFD(file))) {
+    if (file->type == FILE_TYPE_IN_MEMORY) {
         return -EBADF;
     }
 
-    if (file->type == FILE_TYPE_IN_MEMORY) {
+    if (!_fd_isValid(_regularfile_getOSBackedFD(file))) {
```

Comment:
I don't understand why this change is necessary, or the other similar changes below, because I think the code is equivalent with or without the change?

Since the two `if` statements are executed in sequence, isn't it equivalent to

```C
if (file->type == FILE_TYPE_IN_MEMORY || !_fd_isValid(_regularfile_getOSBackedFD(file))) {
    return -EBADF;
}    
```

in which case the order should not matter?

Reply:
Yup! Basically I was moving the `_isValid` checks to just-before the native calls, to better convey/preserve the intent that it's there primarily to avoid a native syscall that we know would fail.